R/looping.R

```diff
@@ -55,65 +51,61 @@ slide_forecaster <- function(data,
     net_slide_training <- slide_training + n_training_pad
   }
   # restrict the dataset to areas where training is possible
-  start_date <- min(archive$DT$time_value) + net_slide_training
-  end_date <- max(archive$DT$time_value) - forecaster_args$ahead
+  start_date <- min(epi_archive$DT$time_value) + net_slide_training
+  end_date <- max(epi_archive$DT$time_value) - forecaster_args$ahead
   valid_predict_dates <- seq.Date(from = start_date, to = end_date, by = date_range_step_size)
 
   # first generate the forecasts
   before <- n_training + n_training_pad - 1
-  ## TODO: epix_slide doesn't support infinite `before`
-  ## https://github.com/cmu-delphi/epiprocess/issues/219
-  if (before == Inf) before <- 365L * 10000
-  res <- epix_slide(archive,
-    function(data, gk, rtv, ...) {
-      # TODO: Can we get rid of this tryCatch and instead hook it up to targets
-      #       error handling or something else?
-      #       https://github.com/cmu-delphi/exploration-tooling/issues/41
-      tryCatch(
-        {
-          do.call(
-            forecaster,
-            append(
-              list(
-                epi_data = data,
-                outcome = outcome,
-                extra_sources = extra_sources
-              ),
-              forecaster_args
-            )
-          )
-        },
-        error = function(e) {
```

Comment:
did you figure out an alternative to this?

Reply:
I think the dummy_forecaster is the alternative. We still need to find a way to automatically have dummy_forecaster replace the actual forecasting step inside a recipe with a simple debug flag (so we can inspect the data after the recipe steps), but I think we can get there incrementally.